File: CodingPractice-master/leetcode/algorithms/cpp/0326_PowerOfThree.cpp
/*
* @Author: Yan_Daojiang
* @Date:   2019-05-17 10:47:51
* @Last Modified by:   Yan_Daojiang
* @Last Modified time: 2019-05-17 10:48:05
*/
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n==0)
            return false;
        while(n!=0){
            if(n%3!=0&&n!=1)
                return false;
            if(n==1)
                return true;
            n/=3;
        }
        return true;
    }
};

File: CodingPractice-master/leetcode/algorithms/python/0326_PowerOfThree.py
# -*- coding: utf-8 -*-
# @Author: Yan_Daojiang
# @Date:   2019-05-17 10:44:55
# @Last Modified by:   Yan_Daojiang
# @Last Modified time: 2019-05-17 10:46:56
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n==0:
            return False
        while n!=0:
            if n%3!=0 and n!=1:
                return False
            if n==1:
                return True;
            n=int(n/3)
        return True